libr/core/cmd_search.c

```diff
@@ -2116,10 +2082,17 @@ void _CbInRangeSearchV(RCore *core, ut64 from, ut64 to, int vsize, bool asterisk
 }
 
 static int cmd_search(void *data, const char *input) {
-	struct search_parameters param;
 	bool dosearch = false;
 	int i, len, ret = true;
 	RCore *core = (RCore *) data;
+	struct search_parameters param = {
```

Comment:
all this stuff should be accessible from core->search, no need to add another abstraction for that

Reply:
In `	r_search_set_callback (rs, &hit, buf);` set callback, `core` is not accessible.